developer-hub/src/reference_docs/extrinsic_encoding.rs

```diff
@@ -0,0 +1,172 @@
+//! [here](https://github.com/polkadot-fellows/runtimes/blob/1dc04eb954eadf8aadb5d83990b89662dbb5a074/relay/polkadot/src/lib.rs#L1432C25-L1432C25).
+//! Some of the common signed extensions are defined [here][frame::deps::frame_system].
+//!
+//! Information about exactly which signed extensions are present on a chain and in what order is
+//! also a part of the metadata for the chain. For V15 metadata, it can be
+//! [found here][frame::deps::frame_support::__private::metadata::v15::ExtrinsicMetadata].
+//!
+//! ## call_data
+//!
+//! This data defines exactly which call is made by the extrinsic, and with what arguments. These
+//! are determined by the second generic parameter of [`sp_runtime::generic::UncheckedExtrinsic`].
+//!
+//! In pseudo-code, a call looks like this:
+//!
+//! ```text
+//! call_data = concat(
+//!     pallet_index,
+//!     call_index,
+//!     call_args
+//! )
+//! ```
+//!
+//! - `pallet_index` is a single byte denoting the index of the pallet that we are calling into.
+//! - `call_index` is a single byte denoting the index of the call that we are making the pallet.
+//! - `call_args` are the SCALE encoded bytes for each of the arguments that the call expects.
+//!
+//! Information about the pallets that exist for a chain (including their indexes), the calls
+//! available in each pallet (including their indexes), and the arguments required for each call
+//! can be found in the metadata for the chain. For V15 metadata, this information
+//! [is here][frame::deps::frame_support::__private::metadata::v15::PalletMetadata].
+//!
+//! # The Signed Payload Format
+//!
+//! All extrinsics submitted to a node from the outside world (also known as _transactions_) need to
+//! be _signed_. The data that needs to be signed for some extrinsic is called the _signed payload_
+//! (also called the _signed payload_), and its shape is described by the following pseudo-code:
+//!
+//! ```text
+//! signed_payload = concat(
+//!     call_data,
+//!     signed_extensions_extra,
+//!     signed_extensions_additional,
+//! )
+//!
+//! if length(signed_payload) > 256 {
+//!     signed_payload = blake2_256(signed_payload)
+//! }
+//! ```
+//!
+//! In other words, we create the signed payload by concatenating the bytes representing
+//! `call_data`, `signed_extensions_extra` and `signed_extensions_additional` together. If this
+//! payload is more than 256 bytes in size, we hash it using a 256bit Blake2 hasher.
+//!
+//! How to construct the `call_data` and `signed_extensions_extra` has already been explained above.
+//! `signed_extensions_additional` is constructed by SCALE encoding the
+//! ["additional signed" data][sp_runtime::traits::SignedExtension::AdditionalSigned] for each signed
+//! extension that the chain is using, in order, and concatenating the resulting bytes together.
+//!
+//! If the resulting bytes have a length greater than 256, then we hash them using a Blake2 256bit
+//! hasher, and that hash is the signed payload.
```

Comment:
All I am kinda itching for is a more practical example, but you know better, and maybe it is an overkill. How about a snippet that creates a very basic instance of `UncheckedExtrinsic`, then `.encode()` it, then try and cluster the output bytes into different sections that you explained in the doc?

Reply:
Yup I think you're right, and I do love examples myself! 

Maybe I can add an example at the end showing how to manually construct the bytes for some extrinsic, and then prove that they are correct (with a nice hidden assert to make sure it stays this way) when using `UncheckedExtrinsic` to do the same!